From 36db55c34d65eb39554259e023e3b640cde9cf12 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Wed, 19 Nov 2008 10:08:14 -0800
Subject: [PATCH 06/21] Staging: comedi: fix sparse issues in comedi_bond.c

This fixes the issues raised by the sparse tool on the
comedi_bond.c file.

Cc: Calin A. Culianu <calin@ajvar.org>
Cc: David Schleef <ds@schleef.org>
Cc: Frank Mori Hess <fmhess@users.sourceforge.net>
Cc: Ian Abbott <abbotti@mev.co.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/comedi/drivers/comedi_bond.c |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

--- a/drivers/staging/comedi/drivers/comedi_bond.c
+++ b/drivers/staging/comedi/drivers/comedi_bond.c
@@ -103,7 +103,7 @@ MODULE_LICENSE("GPL");
 #  define STR(x) STR1(x)
 #endif
 
-int debug;
+static int debug;
 module_param(debug, int, 0644);
 MODULE_PARM_DESC(debug, "If true, print extra cryptic debugging output useful"
 		 "only to developers.");
@@ -408,7 +408,7 @@ static int doDevConfig(comedi_device *de
 		int minor = it->options[i];
 		comedi_t *d;
 		int sdev = -1, nchans, tmp;
-		struct BondedDevice *bdev = 0;
+		struct BondedDevice *bdev = NULL;
 
 		if (minor < 0 || minor > COMEDI_NUM_BOARD_MINORS) {
 			ERROR("Minor %d is invalid!\n", minor);
@@ -515,18 +515,18 @@ static void doDevUnconfig(comedi_device 
 			kfree(bdev);
 		}
 		kfree(devpriv->devs);
-		devpriv->devs = 0;
+		devpriv->devs = NULL;
 		kfree(devpriv);
-		dev->private = 0;
+		dev->private = NULL;
 	}
 }
 
-int __init init(void)
+static int __init init(void)
 {
 	return comedi_driver_register(&driver_bonding);
 }
 
-void __exit cleanup(void)
+static void __exit cleanup(void)
 {
 	comedi_driver_unregister(&driver_bonding);
 }
